### PR TITLE
Setup Baseline Profile

### PR DESCRIPTION
**Pull request by ivaniskandar**
Adds Baseline Profile generator and startup time test. Readme included in macrobenchmark module to run the generator.

```
ColdStartupBenchmark_startupNoCompilation
timeToInitialDisplayMs   min 788.8,   median 804.0,   max 965.1

ColdStartupBenchmark_startupFullCompilation
timeToInitialDisplayMs   min 758.3,   median 783.0,   max 820.0

ColdStartupBenchmark_startupBaselineProfile
timeToInitialDisplayMs   min 731.1,   median 736.2,   max 768.8

ColdStartupBenchmark_startupBaselineProfileDisabled
timeToInitialDisplayMs   min 746.2,   median 769.5,   max 787.7
```

### CODE REVIEW

File: gradle/androidx.versions.toml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 [versions]
 agp_version = "7.3.0"
 lifecycle_version = "2.5.1"
+androidxEspresso = "3.4.0"
```
Comment (arkon) [inline gradle/androidx.versions.toml:4]: The `androidx` prefixes seem redundant given the file they're in.

File: macrobenchmark/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+plugins {
+    id("com.android.test")
+    id("org.jetbrains.kotlin.android")
```
Comment (arkon) [inline macrobenchmark/build.gradle.kts:3]:
```suggestion
    id("org.jetbrains.kotlin.android")
    kotlin("android")
```

I think

File: macrobenchmark/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    defaultConfig {
+        minSdk = AndroidConfig.minSdk
+        targetSdk = AndroidConfig.targetSdk
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        // This benchmark buildType is used for benchmarking, and should function like your
+        // release build (for example, with minification on). It"s signed with a debug key
```
Comment (arkon) [inline macrobenchmark/build.gradle.kts:28]: Apostrophe

File: macrobenchmark/src/main/java/tachiyomi/macrobenchmark/BaselineProfileGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package tachiyomi.macrobenchmark
+
+import androidx.benchmark.macro.ExperimentalBaselineProfilesApi
+import androidx.benchmark.macro.junit4.BaselineProfileRule
+import org.junit.Rule
+import org.junit.Test
```
Comment (arkon) [inline macrobenchmark/src/main/java/tachiyomi/macrobenchmark/BaselineProfileGenerator.kt:6]: I'm guessing there's no junit 5 support still...

File: macrobenchmark/src/main/java/tachiyomi/macrobenchmark/BaselineProfileGenerator.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Test
+    fun generate() = baselineProfileRule.collectBaselineProfile(
+        packageName = "eu.kanade.tachiyomi.benchmark",
+        profileBlock = {
+            pressHome()
+            startActivityAndWait()
+
+            // TODO: Navigate to browse-extensions screen when storage permission
+            // in sources screen moved. Possibly open manga details screen too?
```
Comment (arkon) [inline macrobenchmark/src/main/java/tachiyomi/macrobenchmark/BaselineProfileGenerator.kt:21]: Can we seed the db or something?

Comment (ivaniskandar) [reply macrobenchmark/src/main/java/tachiyomi/macrobenchmark/BaselineProfileGenerator.kt:21]: Maybe, still looking for a proper solution as I'm not familiar with db testing/mocking

**Pull request merged**